tests/ui/lint/dropping_references-can-fixed.stderr

```diff
@@ -45,7 +45,7 @@ LL -     drop(&&owned1);
 LL +     let _ = &&owned1;
    |
 
-error: calls to `std::mem::drop` with a reference instead of an owned value does nothing
```

Comment:
So it's the "does nothing" part that is the problem here? I.e. it's not true for mutable references?

How about just changing the message to this: "a call to `std::mem::drop` that is passed a reference drops the reference, not the underlying value". It is true for both kinds of reference, and avoids the need for two different error messages?

Reply:
> So it's the "does nothing" part that is the problem here? I.e. it's not true for mutable references?

Yes.

> How about just changing the message to this: "a call to `std::mem::drop` that is passed a reference drops the reference, not the underlying value". It is true for both kinds of reference, and avoids the need for two different error messages?

Works for me, I'm not attached to the original message.

Reply:
> > So it's the "does nothing" part that is the problem here? I.e. it's not true for mutable references?
> 
> Yes.

IMO, this isn't true, the call to `std::mem::drop` with a reference (mutable or not) does nothing *to the reference or the underline value*. The fact that is does something to the binding/variable it-self is unfortunate but IMO irrelevant here.

------

> How about just changing the message to this: "a call to `std::mem::drop` that is passed a reference drops the reference, not the underlying value".

I would personally prefer: "calls to `std::mem::drop` with a reference does not drop the underline value"